std/machines/arith.asm

```diff
@@ -132,66 +132,38 @@ machine Arith with
         0
     };
 
-    let quotient_hint = query || {
-        let y2 = y2_int();
-        let y3 = y3_int();
-        let x1 = x1_int();
-        let dividend = (y2 << 256) + y3;
-        let quotient = dividend / x1;
-        quotient
-    };
-
-    let remainder_hint = query || {
-        let y2 = y2_int();
-        let y3 = y3_int();
-        let x1 = x1_int();
-        let dividend = (y2 << 256) + y3;
-        let remainder = dividend % x1;
-        remainder
-    };
-
-    let hint_if_eq0 = query |f, limb| match is_ec_operation() {
-        0 => Query::Hint(fe(select_limb(f(), limb))),
-        _ => Query::None
+    col witness y1[16];
+    col witness x2[16];
+    query |i| {
+        match is_ec_operation() {
+            0 => {
+                match std::prover::try_eval(y1[0]) {
+                    Option::Some(_) => {
+                        // y1 is an input, in this case we do not need a hint.
+                    },
```

Comment:
This match block is new to this PR. Did you add this because when `y1` is already provided, `provide_value` will error out? When the operation is `affine_256`, `is_ec_operation = 0` but `y1` is provided as an input, so these hints are only used for `mod_256`.

Reply:
Yes, that is the reason. This is just a first step, the full change of arith.asm is in https://github.com/powdr-labs/powdr/pull/1760